SIPS/sip-20.md

```diff
@@ -0,0 +1,155 @@
+---
+sip: 20
+title: External data (WebSockets and other custom data or events)
+status: Draft
+author: David Drazic (@david0xd)
+created: 2023-12-15
+---
+
+## Abstract
+
+This SIP proposes changes to the Snap manifest and additional request handler that allows Snap developers to implement
+functionality that is invoked when external data events occur. This proposal outlines some of the details around this feature.
+
+The purpose of the initial version of this SIP is to provide support for WebSockets.
+This SIP only proposes one-way communication.
+This SIP is made with extensibility in mind and can be updated to handle more external data sources (like blockchain
+events, etc.).
```

Comment:
```suggestion
This SIP proposes a new permission that enables Snaps to receive data from
external sources. In the initial version, this SIP proposes support for
WebSockets, but it can be extended to support other data sources like
blockchain events in the future.

The Snap can specify the external data source in the Snap manifest. The
client then connects to the external data source and sends the data to the Snap.
The Snap can then do whatever it wants with the data.

This initial version only supports one-way communication from the external
source to the Snap. The Snap can't send any data back to the external source.
```

Reply:
Updated.